File: src/js/main/commands/RichJson_copy.js
/*
    makes a deep copy of the given struct or array
*/

import {cloneObject} from "../RichJsonHelper.js";
import {__executeRefCommand} from "./RichJson_ref.js";

export function __executeCopyCommand(parser, context) {
    return cloneObject(__executeRefCommand(parser, context));
}

File: src/c++/main/commands/RichJson_copy.cpp
#include "RichJson_copy.hpp"
#include "../core/RichJsonCommandHolder.hpp"
#include "../helper/RichJsonHelper.hpp"

namespace RichJson {

json richJson_copy(RichJsonParser& parser, RichJsonContext& context) {
    json referencedValue = RichJsonCommandHolder::executeCommand("ref", parser, context);
    return RichJsonHelper::cloneObject(referencedValue);
}

} // namespace RichJson
